pkg/server/server.go

```diff
@@ -99,65 +110,27 @@ func (s *AdmissionWebhookServer) listenAndServe(clients *clients.Clients, handle
 		// Sleep here to make sure server is listening and all caches are primed
 		time.Sleep(15 * time.Second)
 
-		logrus.Debugf("Building validation rules...")
-		validationRules := s.buildRules(validationResources)
-		logrus.Debugf("Building mutation rules...")
-		mutationRules := s.buildRules(mutationResources)
-
-		validatingWebhookConfiguration := &v1.ValidatingWebhookConfiguration{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: s.name,
-			},
-			Webhooks: []v1.ValidatingWebhook{
-				{
-					Name: "validator." + s.options.Namespace + "." + s.name,
-					ClientConfig: v1.WebhookClientConfig{
-						Service: &v1.ServiceReference{
-							Namespace: s.options.Namespace,
-							Name:      s.name,
-							Path:      &validationPath,
-							Port:      &port,
-						},
-						CABundle: secret.Data[corev1.TLSCertKey],
-					},
-					Rules:                   validationRules,
-					FailurePolicy:           &failPolicyFail,
-					SideEffects:             &sideEffectClassNone,
-					AdmissionReviewVersions: []string{"v1", "v1beta1"},
-				},
-			},
+		s.caBundle = secret.Data[corev1.TLSCertKey]
+		// configure admitter webhook
+		validatingWebhookConfiguration := s.validatingWebhookConfiguration()
+		mutatingWebhookConfiguration := s.mutatingWebhookConfiguration()
+		if validatingWebhookConfiguration != nil || mutatingWebhookConfiguration != nil {
+			if err := apply.WithOwner(secret).ApplyObjects(validatingWebhookConfiguration, mutatingWebhookConfiguration); err != nil {
```

Comment:
should `configureCRDConversionWebhook` related configuration also be applied to `secret` ?

Reply:
No, we configure CRD conversion webhook in the CRD.